Test failure after initial setup for development

### System Info

OS version: Windows 10 Pro
Python version: 3.10.10
PandasAI version: 1.3.3

### 🐛 Describe the bug

After a clean setup for initial development, notice that 2 of the tests are failing:

```bash
================================================================================ short test summary info ================================================================================= 
FAILED tests/test_smartdataframe.py::TestSmartDataframe::test_save_chart_non_default_dir - AssertionError: assert 'exports/char...b180ab5a5.png' == 'exports/char...b180ab5a5.png'
FAILED tests/callbacks/test_file.py::TestFileCallback::test_on_code - PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\Henrique\\Desktop\\area de trabalho\\pessoal\\repos\\pandas...
================================================================= 2 failed, 248 passed, 2 skipped, 40 warnings in 12.09s =================================================================
```

I'll give extra guidance on both errors.

## `test_on_code`
This is returning `none` for the `self.assertEqual(contents, response)` statements, although the strings compared are equal.

And why are mixed frameworks for testing? There are some tests using the built-in `unittest` module. Shouldn't it be standardized all to `pytest`?

## `test_save_chart_non_default_dir`

This may have a connection with the OS. The test is failing on the following assertion:
```python
        assert (
            plt_mock.savefig.call_args.args[0]
            == f"exports/charts/{smart_dataframe.last_prompt_id}.png"
        )
```

This is the best practice because the path is fixed for Unix-style systems. The problem is precisely in the fixed path variable.

```python
plt_mock.savefig.call_args.args[0] = 'exports/charts\\cf69927e-2ec6-4ae6-a063-0d3d30598328.png'
f"exports/charts/{smart_dataframe.last_prompt_id}.png" = 'exports/charts/cf69927e-2ec6-4ae6-a063-0d3d30598328.png'
```

@gventuri, would you like to let someone work on this? Or can I handle it? I prefer to let new contributors have a chance to work on it, especially the second one, which is a path and OS related problem.

--- a/pandasai/connectors/airtable.py
+++ b/pandasai/connectors/airtable.py
@@ -181,10 +181,7 @@ class AirtableConnector(BaseConnector):
         Fetches data from the Airtable server via API and converts it to a DataFrame.
         """
 
-        params = {
-            "pageSize": 100,
-            "offset": "0"
-        }
+        params = {"pageSize": 100, "offset": "0"}
 
         if self._config.where is not None:
             params["filterByFormula"] = self._build_formula()

--- a/pandasai/helpers/code_manager.py
+++ b/pandasai/helpers/code_manager.py
@@ -210,7 +210,7 @@ class CodeManager:
                 code,
                 logger=self._logger,
                 file_name=str(prompt_id),
-                save_charts_path=self._config.save_charts_path,
+                save_charts_path_str=self._config.save_charts_path,
             )
 
         # Get the code to run removing unsafe imports and df overwrites

--- a/pandasai/helpers/save_chart.py
+++ b/pandasai/helpers/save_chart.py
@@ -1,13 +1,13 @@
 """Helper functions to save charts to a file, if plt.show() is called."""
+from pathlib import Path
 from .logger import Logger
-from os import makedirs, path
 
 
 def add_save_chart(
     code: str,
     logger: Logger,
     file_name: str,
-    save_charts_path: str = None,
+    save_charts_path_str: str = None,
 ) -> str:
     """
     Add line to code that save charts to a file, if plt.show() is called.
@@ -22,12 +22,15 @@ def add_save_chart(
         str: Code with line added.
 
     """
-    if save_charts_path is not None:
-        if not path.exists(save_charts_path):
-            makedirs(save_charts_path)
 
-        save_charts_file = path.join(save_charts_path, file_name + ".png")
-        code = code.replace("temp_chart.png", save_charts_file)
+    save_charts_path = Path(save_charts_path_str) if save_charts_path_str else None
+
+    if save_charts_path is not None:
+        save_charts_path.mkdir(parents=True, exist_ok=True)
+
+        save_charts_file = save_charts_path / f"{file_name}.png"
+
+        code = code.replace("temp_chart.png", save_charts_file.as_posix())
         logger.log(f"Saving charts to {save_charts_file}")
 
     return code

--- a/pandasai/llm/google_vertexai.py
+++ b/pandasai/llm/google_vertexai.py
@@ -129,7 +129,6 @@ class GoogleVertexAI(BaseGoogle):
         else:
             raise UnsupportedModelError(self.model)
 
-
         return str(completion)
 
     @property

--- a/pandasai/llm/openai.py
+++ b/pandasai/llm/openai.py
@@ -109,7 +109,6 @@ class OpenAI(BaseOpenAI):
         else:
             raise UnsupportedModelError(self.model)
 
-
         return response
 
     @property

--- a/tests/callbacks/test_file.py
+++ b/tests/callbacks/test_file.py
@@ -10,6 +10,7 @@ class TestFileCallback(unittest.TestCase):
 
     def tearDown(self):
         if os.path.exists(self.filename):
+            self.callback.file.close()
             os.remove(self.filename)
 
     def test_on_code(self):

--- a/tests/connectors/test_airtable.py
+++ b/tests/connectors/test_airtable.py
@@ -13,9 +13,7 @@ class TestAirTableConnector(unittest.TestCase):
             api_key="your_token",
             base_id="your_baseid",
             table="your_table_name",
-            where= [
-                ["Status", "=", "In progress"]
-            ]
+            where=[["Status", "=", "In progress"]],
         ).dict()
         self.root_url = "https://api.airtable.com/v0/"
         self.expected_data_json = """
@@ -104,14 +102,14 @@ class TestAirTableConnector(unittest.TestCase):
         mock_request_get.return_value.status_code = 200
         columns_count = self.connector.columns_count
         self.assertEqual(columns_count, 3)
-    
+
     def test_build_formula_method(self):
         formula = self.connector._build_formula()
         expected_formula = "AND(Status='In progress')"
-        self.assertEqual(formula,expected_formula)
-    
+        self.assertEqual(formula, expected_formula)
+
     @patch("requests.get")
-    def test_column_hash(self,mock_request_get):
+    def test_column_hash(self, mock_request_get):
         mock_request_get.return_value.json.return_value = json.loads(
             self.expected_data_json
         )
@@ -119,5 +117,5 @@ class TestAirTableConnector(unittest.TestCase):
         returned_hash = self.connector.column_hash
         self.assertEqual(
             returned_hash,
-            "e4cdc9402a0831fb549d7fdeaaa089b61aeaf61e14b8a044bc027219b2db941e"
+            "e4cdc9402a0831fb549d7fdeaaa089b61aeaf61e14b8a044bc027219b2db941e",
         )

--- a/tests/helpers/test_save_chart.py
+++ b/tests/helpers/test_save_chart.py
@@ -1,4 +1,5 @@
 import os
+from pathlib import Path
 import tempfile
 from pandasai.helpers.logger import Logger
 from pandasai.helpers.save_chart import add_save_chart
@@ -34,13 +35,14 @@ plt.show()
         file_name = "temp_chart"
 
         with tempfile.TemporaryDirectory() as temp_dir:
-            save_charts_path = os.path.join(temp_dir, "charts")
-            assert not os.path.exists(save_charts_path)
+            save_charts_path = Path(temp_dir) / "charts"
+            assert not save_charts_path.exists()
+            full_path_to_file = save_charts_path / f"{file_name}.png"
 
             expected_code = f"""
 import matplotlib.pyplot as plt
 plt.plot([1, 2, 3], [4, 5, 6])
-plt.savefig("{os.path.join(save_charts_path, file_name)}.png")
+plt.savefig("{full_path_to_file.as_posix()}")
 plt.show()
 """
             result = add_save_chart(code, logger, file_name, save_charts_path)

